docs/rules/unused-prop-types.md

```diff
@@ -0,0 +1,70 @@
+# Prevent definitions of unused prop types (unused-prop-types)
+
+Warns you if you have defined a prop type but it is never being used anywhere.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+var Hello = React.createClass({
+  propTypes: {
+    name: React.PropTypes.string
+  },
+  render: function() {
+    return <div>Hello Bob</div>;
+  }
+});
+
+var Hello = React.createClass({
+  propTypes: {
+    firstname: React.PropTypes.string.isRequired,
+    middlename: React.PropTypes.string.isRequired, // middlename is never used below
+	lastname: React.PropTypes.string.isRequired
+  },
+  render: function() {
+    return <div>Hello {this.props.firstname} {this.props.lastname}</div>;
+  }
+});
+
+function Hello({ name }) {
+  return <div>Hello Jeremy</div>;
+}
+```
+
+The following patterns are not considered warnings:
+
+```jsx
+var Hello = React.createClass({
+  propTypes: {
+    name: React.PropTypes.string
+  },
+  render: function() {
+    return <div>Hello {this.props.name}</div>;
+  }
+});
+```
+
+## Rule Options
+
+This rule can take one argument to ignore some specific props during validation.
+
+```
+...
+"prop-types": [<enabled>, { customValidators: <customValidator> }]
+...
+```
+
+* `enabled`: for enabling the rule. 0=off, 1=warn, 2=error. Defaults to 0.
+* `customValidators`: optional array of validators used for propTypes validation.
```

Comment:
To be honest with you, I don't know of good use cases for it either. I've included it only because the `prop-types` rule also has this option. And its documentation is exactly as this.